ENYO-3260 : Change read sequence in popup, contexturePopup

changgilee opened the pull request:
In current status, when popup(or contexturePopup) is shown, read "button" + "title" sequence.

But UX guideline in Major MR, "title" + "button" sequence is right.

When onSpotlightContainerEnter is fired, the button in popup is not focused yet.

So, if change role to alert in this handler, "alert" + "dom.focus()" rule is applied and TTS call is not cut off.

Enyo-DCO-1.1-Signed-off-by: Changgi Lee changgi.lee@lge.com

changgilee committed:
ENYO-3260 : Change read sequence in popup, contexturePopup

In current status, when popup(or contexturePopup) is shown, read <button> + <title> sequence.

But UX guideline in Major MR, <title> + <button> sequence is right.

When onSpotlightContainerEnter is fired, the button is not focused yet.

So, if change role to alert in this handler, <alert> + <focus> rule is applied and TTS call is not cut off.

Enyo-DCO-1.1-Signed-off-by: Changgi Lee <changgi.lee@lge.com>

kbs12e commented on src/Popup/Popup.js:
```diff
@@ -596,5 +597,14 @@ module.exports = kind(
+	* @private
+	*/
+	onEnter : function(oSender, oEvent){
+		if (oEvent.originator == this){
```
I think that accessibility option checking is neccessary.

changgilee replied:
I will add it~ :)

kbs12e commented on src/ContextualPopup/ContextualPopup.js:
```diff
@@ -659,5 +660,14 @@ module.exports = kind(
+	* @private
+	*/
+	onEnter : function(oSender, oEvent){
+		if (oEvent.originator == this){
```
accessibility option checking is neccessary

changgilee replied:
I will add it~ :)

kbs12e commented on src/Popup/Popup.js:
```diff
@@ -596,5 +597,14 @@ module.exports = kind(
 				this.setAriaAttribute('role', this.accessibilityReadAll && this.showing ? 'alert' : this.accessibilityRole);
 			}, 100);
```
Basic popup, which has only content doesn't needs timer delay.

changgilee replied:
You are right~ I will remove timer delay~

kbs12e commented on src/Popup/Popup.js:
```diff
@@ -596,5 +597,14 @@ module.exports = kind(
 				this.setAriaAttribute('role', this.accessibilityReadAll && this.showing ? 'alert' : this.accessibilityRole);
```
This line is similar to onEnter() callback implementation.

changgilee replied:
Can you give me more guidance what is wrong in this code? Thanks~

kbs12e committed:
ENYO-3260: refactor onEnter()

Add new _enableAlert property for checking onSpotlightContainerEnter event.

https://jira2.lgsvl.com/browse/ENYO-3260
Enyo-DCO-1.1-Signed-off-by: Bongsub Kim <bongsub.kim@lgepartner.com>

Change-Id: I0c8d20a9d85ab1664c1ea9a90edb05647a04067a

changgilee committed:
ENYO-3260 : Add updateAriaRole function

Add updateAriaRole function for reduce duplicate code

Enyo-DCO-1.1-Signed-off-by: Changgi Lee <changgi.lee@lge.com>

kbs12e commented on src/Popup/Popup.js:
```diff
@@ -427,6 +430,7 @@ module.exports = kind(
 	* @public
 	*/
 	hide: function() {
+		this.set('_enableAlert', false);
```
If updateAriaRole() is called, this line is not neccessary.

kbs12e commented on src/Popup/Popup.js:
```diff
@@ -591,10 +595,24 @@ module.exports = kind(
-			this.startJob('alert', function () {
-				this.setAriaAttribute('role', this.accessibilityReadAll && this.showing ? 'alert' : this.accessibilityRole);
-			}, 100);
+		{path: ['accessibilityReadAll', 'accessibilityRole', 'showing', '_enableAlert'], method: function () {
```
If updateAriaRole() is called, _enableAlert observer is not neccessary.

kbs12e committed:
ENYO-3260: remove unneccessary code and timer delay from tooltip.

Remove unneccessary code.
Remove timer delay code from moon.Tooltip, because its tooltip is popup type, which
replaced timer code with direct function call.

https://jira2.lgsvl.com/browse/ENYO-3260
Enyo-DCO-1.1-Signed-off-by: Bongsub Kim <bongsub.kim@lgepartner.com>

Change-Id: I7b550190bca85f1af6de441c3d62f3e2329cd5d4

lishammel commented:
Closing PR as Jira has been closed.
